src/geometries/Geometries.js

```diff
@@ -19,3 +19,4 @@ export * from './TorusGeometry.js';
 export * from './TorusKnotGeometry.js';
 export * from './TubeGeometry.js';
 export * from './WireframeGeometry.js';
+export * from '../../examples/jsm/geometries/TextGeometry.js';
```

Comment:
I don't think we can refer in `src` (the core) to addon files. Would it work somehow without this line?

Reply:
We have moved `FontLoader` and `TextGeometry` out of the core since they are not very common classes. So I don't think adding them back to core would be good.

I wonder if we could somehow register a type for deserialization in `ObjectLoader`. Something like:
```js
const loader = new THREE.ObjectLoader();
loader.registerType( 'TextGeometry', TextGeometry );
```
This could be done manually in the editor before the scene is loaded/paresed.

Reply:
Yeah there's probably 2 ways to around this

1. moving TextGeometry from to core while FontLoader can remain in addons
2. dynamically adding the registered list of Geometry types


I can understand the rationale for wanting to moving TextGeometry back to core, but for what you're describing in 2, there're 3 places in editor where ObjectLoader is used, which we could update all places that's called or perhaps have another ways to update the static list everywhere? Perhaps


Geometries.registerType( 'TextGeometry', TextGeometry );

or something?

Reply:
This actually seems very similar to how the Nodes system does a similar thing -- namely, `addNodeClass`, `addNodeElement`, `addLightNode`, `addNodeMaterial`.

Reply:
> moving TextGeometry from to core while FontLoader can remain in addons

It seems `TextGeometry` has this import now:
```js
import { Font } from '../loaders/FontLoader.js';
```
This does not allow to move `TextGeometry` into core without `FontLoader` otherwise have again a direct dependency to addons.

So we should try to find a solution for suggestion 2.

Reply:
> It seems TextGeometry has this import now:

gotcha. Turns out I'll only register the TextGeometry class under Editor.js so I think I'll go with

```js
		var loader = new THREE.ObjectLoader();
		loader.registerGeometry( 'TextGeometry', TextGeometry );
```

Reply:
This looks great to me! 👍